src/xo-app/vm/tab-disks.js

```diff
@@ -50,6 +51,127 @@ import {
   subscribeResourceSets,
 } from 'xo'
 
+const COLUMNS_VM_PV = [
+  {
+    itemRenderer: vdi => (
+      <Text
+        value={vdi.name_label}
+        onChange={value => editVdi(vdi, { name_label: value })}
+      />
+    ),
+    name: _('vdiNameLabel'),
+    sortCriteria: 'name_label',
+    default: true,
+  },
+  {
+    itemRenderer: vdi => (
+      <Text
+        value={vdi.name_description}
+        onChange={value => editVdi(vdi, { name_description: value })}
+      />
+    ),
+    name: _('vdiNameDescription'),
+    sortCriteria: 'name_description',
+  },
+  {
+    itemRenderer: vdi => (
+      <Size
+        value={vdi.size || null}
+        onChange={size => editVdi(vdi, { size })}
+      />
+    ),
+    name: _('vdiSize'),
+    sortCriteria: 'size',
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const sr = userData.srs[vdi.$SR]
+      return (
+        sr !== undefined && (
+          <XoSelect
+            labelProp='name_label'
+            onChange={sr => migrateVdi(vdi, sr)}
+            predicate={sr => sr.$pool === userData.vm.$pool && isSrWritable(sr)}
+            useLongClick
+            value={sr}
+            xoType='SR'
+          >
+            <Link to={`/srs/${sr.id}`}>{sr.name_label}</Link>
+          </XoSelect>
+        )
+      )
+    },
+    name: _('vdiSr'),
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbd = find(userData.vbds, vbd => vbd.id === vdi.$VBDs[0])
+      return (
+        <Toggle
+          onChange={bootable => setBootableVbd(vbd, bootable)}
+          value={vbd.bootable}
+        />
+      )
+    },
+    name: _('vbdBootableStatus'),
+    id: 'vbdBootableStatus',
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbd = find(userData.vbds, vbd => vbd.id === vdi.$VBDs[0])
+      return (
+        <StateButton
+          disabledLabel={_('vbdStatusDisconnected')}
+          disabledHandler={connectVbd}
+          disabledTooltip={_('vbdConnect')}
+          enabledLabel={_('vbdStatusConnected')}
+          enabledHandler={disconnectVbd}
+          enabledTooltip={_('vbdDisconnect')}
+          disabled={!(vbd.attached || isVmRunning(userData.vm))}
+          handlerParam={vbd}
+          state={vbd.attached}
+        />
+      )
+    },
+    name: _('vbdStatus'),
+  },
+]
+
+const COLUMNS = filter(COLUMNS_VM_PV, col => col.id !== 'vbdBootableStatus')
+
+const FILTERS = {
+  filterOnlyManaged: 'type:!VDI-unmanaged',
+  filterOnlyRegular: '!type:|(VDI-snapshot VDI-unmanaged)',
+  filterOnlySnapshots: 'type:VDI-snapshot',
+  filterOnlyOrphaned: 'type:!VDI-unmanaged $VBDs:!""',
+  filterOnlyUnmanaged: 'type:VDI-unmanaged',
+}
+
+const GROUPED_ACTIONS = [
+  {
+    disabled: (selectedItems, userData) => {
+      const vbds = filter(userData.vbds, vbd =>
+        find(selectedItems, vdi => vdi.$VBDs[0] === vbd.id)
```

Comment:
I'm not sure what you're trying to do here. Anyway, the `filter`'s callback should return a boolean.